awesomeConfig.cmake

```diff
@@ -371,7 +371,26 @@ add_custom_command(
         OUTPUT ${BUILD_DIR}/docs/06-appearance.md
         COMMAND lua ${SOURCE_DIR}/docs/06-appearance.md.lua
         ${BUILD_DIR}/docs/06-appearance.md
-        DEPENDS lgi-check-run ${SOURCE_DIR}/docs/06-appearance.md.lua
+        DEPENDS
+            lgi-check-run
+            ${SOURCE_DIR}/docs/06-appearance.md.lua
+            ${SOURCE_DIR}/docs/_parser.lua
+)
+
+add_custom_command(
+        OUTPUT ${BUILD_DIR}/docs/common/rules_index.ldoc
+        COMMAND lua ${SOURCE_DIR}/docs/build_rules_index.lua
+            ${BUILD_DIR}/docs/common/rules_index.ldoc
+
+        # Cheap trick until the ldoc `configure_file` is ported to be a build
+        # step rather than part of cmake.
+        COMMAND ${CMAKE_COMMAND} -E copy ${BUILD_DIR}/docs/common/rules_index.ldoc
+            ${SOURCE_DIR}/docs/common/rules_index.ldoc
```

Comment:
Uhm... Does this copy something into the *source* directory? Why?!?
Isn't the build supposed not to touch the source directory in any way? Autoconf even checks for that during `make distcheck` by testing an out-of-tree build against read-only sources.

Reply:
Not pretty, I will agree on that. The problem is that `configure_file` depends on it and fixing that will be non-trivial. tl;dr, we will need a `configure_file` equivalent implemented in Lua. It is worth doing and generally would make sense anyway. However I chose not to do it in this pull request as I want v4.3 released and this is part 2 of 4 of the "improve spawn" serie.